### PR TITLE
Feature/oe 237 acordeon table component

### PR DESCRIPTION
**Pull request by Ron-Junior**
## Review

For the submitter, initial self-review:

- [ ] Reviewed the test cases added for completeness and possible points for discussion
- [ ] Check the scope of the changeset - is everything in the diff required for the pull request?

For the pull request reviewer(s), this changeset has been reviewed for:

- [ ] Consistency between the changeset and the goal stated above
- [ ] Usage friction - is the proposed change in usage cumbersome or complicated?
- [ ] Complexity - is the proposed change introducing any unjustified complexity? 
- [ ] Thoroughness of added tests and any missing edge cases

### CODE REVIEW

File: app/Http/Livewire/Components/NumberTrackerDetailAccordionTable.php [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    public Carbon $endDate;
+
+    public function mount()
+    {
+        $query = Region::with(['offices', 'offices.users']);
+
+        if (user()->hasAnyRole(['Admin', 'Owner'])) {
+            $this->regions = $query->get();
+        } else {
+            $this->regions = $query->whereDepartmentid(user()->department_id ?? 0);
```
Comment (devsquad-mathews-lima) [inline app/Http/Livewire/Components/NumberTrackerDetailAccordionTable.php:30]: Here have 2 errors.

1. it should be ```whereDepartmentId```
2. it should have a ```get()``` in the end, because by now you're returning a ```Illuminate\Database\Eloquent\Builder ```.

File: app/Http/Livewire/Components/NumberTrackerDetailAccordionTable.php [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+    public array $itsOpenRegions;
+
+    public Carbon $startDate;
+
+    public Carbon $endDate;
+
+    public function mount()
+    {
+        $query = Region::with(['offices', 'offices.users']);
```
Comment (devsquad-mathews-lima) [inline app/Http/Livewire/Components/NumberTrackerDetailAccordionTable.php:25]: Since regions are not being mutated, I think that is a good chance to use it as a computed property.

File: resources/views/components/table/tr-accordion.blade.php [hunk 3]
```diff
@@ -0,0 +1,16 @@
+@props(['loop', 'index'])
+
+<table class="w-6 h-6 bg-green-600" x-data="{
+                open: true,
+                colapseRaw () {
+                    console.log('teste')
+                }
+            }" >
```
Comment (devsquad-mathews-lima) [inline resources/views/components/table/tr-accordion.blade.php:8]: The x-data container will still be used? If so, don't forget to remove the console.log.

**Review by devsquad-mathews-lima — CHANGES_REQUESTED** [review]

**Review by devsquad-mathews-lima — APPROVED** [review]
